src/Dfc.ProviderPortal.TribalExporter/Functions/ProviderMigrator.cs

```diff
@@ -0,0 +1,361 @@
+﻿using CsvHelper;
+using Dfc.CourseDirectory.Models.Models.Providers;
+using Dfc.ProviderPortal.Packages.AzureFunctions.DependencyInjection;
+using Dfc.ProviderPortal.TribalExporter.Interfaces;
+using Dfc.ProviderPortal.TribalExporter.Models.Tribal;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.Storage.Blob;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UkrlpService;
+
+namespace Dfc.ProviderPortal.TribalExporter.Functions
+{
+    public static class ProviderMigrator
+    {
+        [FunctionName(nameof(ProviderMigrator))]
+        [NoAutomaticTrigger]
+        public static async Task Run(
+                    string input,  // Work around https://github.com/Azure/azure-functions-vs-build-sdk/issues/168
+                    ILogger log,
+                    [Inject] IConfigurationRoot configuration,
+                    [Inject] IProviderCollectionService providerCollectionService,
+                    [Inject] ICosmosDbHelper cosmosDbHelper,
+                    [Inject] IBlobStorageHelper blobhelper,
+                    [Inject] IUkrlpApiService ukrlpApiService
+                    )
+        {
+            const string WHITE_LIST_FILE = "ProviderWhiteList.txt";
+            const string ProviderAppName = "Provder.Migrator";
+
+            var stopWatch = new Stopwatch();
+
+            // TODO : Change to correct collection below
+            var databaseId = configuration["CosmosDbSettings:DatabaseId"];
+            var providerCollectionId = configuration["CosmosDbCollectionSettings:ProvidersCollectionId"];
+            var connectionString = configuration.GetConnectionString("DefaultConnection");
+            var venueExportFileName = $"ProviderExport-{DateTime.Now.ToString("dd-MM-yy HHmm")}";
+
+            var blobContainer = blobhelper.GetBlobContainer(configuration["BlobStorageSettings:Container"]);
+
+            log.LogInformation($"WhitelistProviders : Start reading...");
+            stopWatch.Start();
+            var whiteListProviders = await GetProviderWhiteList();
+            stopWatch.Start();
```

Comment:
should this be stopWatch.Stop(), if your not overly precious on timing fetching the whitelist file, i wouldn't worry too much about changing it.